### PR TITLE
feat(useFocusTrap): new function

### LINKED ISSUE #420: useFocusTrap: new function
This is a proposal that would live under `@vueuse/integrations` as `useFocusTrap`. It would be a wrapper around https://github.com/focus-trap/focus-trap

My proposed api would look something like although I'm not 100% set on the names for the methods yet. Let me know if you think this would be a good addition and I'll open a PR

```ts
import { ref } from 'vue'
import { useFocusTrap } from '@vueuse/integrations'

const container = ref()
const { isTrapped, activate, deactivate  } = useFocusTrap(container, { options })
```

### PR DESCRIPTION
**Pull request by wheatjs**
Implementation of `useFocusTrap` function as discussed in #420.

Closes #420

### CODE REVIEW

File: packages/integrations/useFocusTrap/index.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+---
+category: '@Integrations'
+---
+
+# useFocusTrap
+
+Reactive wrapper for [`useFocusTrap`](https://github.com/focus-trap/focus-trap)
```
Comment (antfu) [inline packages/integrations/useFocusTrap/index.md:7]:
```suggestion
Reactive wrapper for [`focus-trap`](https://github.com/focus-trap/focus-trap)
```

File: packages/integrations/useFocusTrap/index.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+   *
+   * @link https://github.com/focus-trap/focus-trap#trapunpause
+   */
+  unpause: Fn
+}
+
+/**
+ * Reactive focus-trap
+ *
+ * @link https://vueuse.org/integrations/useFocusTrap/
```
Comment (antfu) [inline packages/integrations/useFocusTrap/index.ts:57]:
```suggestion
 * @link https://vueuse.org/useFocusTrap
```

We have HTTP alias set in Netlify

File: packages/integrations/useFocusTrap/index.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      trap.unpause()
+      isPaused.value = false
+    }
+  }
+
+  const stop = watch(
+    () => unrefElement(target),
+    (el: HTMLElement) => {
+      if (el)
+        stop()
```
Comment (antfu) [inline packages/integrations/useFocusTrap/index.ts:90]: Should we support continue changing targets as other functions do?

File: packages/integrations/useFocusTrap/demo.vue [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+</script>
+
+<template>
+  <div class="flex flex-col items-center">
+    <button @click="activate()">
+      Trap Focus
+    </button>
+    <input
+      type="text"
+      :placeholder="hasFocus ? 'You can\'t foucs me' : 'You can focus me'"
```
Comment (antfu) [inline packages/integrations/useFocusTrap/demo.vue:16]: The demo looks interesting 👍

Maybe a line of "The focus is trapped in the form" somewhere could better showing the feature?

File: packages/integrations/useFocusTrap/index.ts (hunk 3 above)
Comment (wheatjs) [reply packages/integrations/useFocusTrap/index.ts:90]: Hmm yeah that's not a bad idea. I'll see what I can do

File: packages/integrations/useFocusTrap/index.ts [hunk 5]
```diff
@@ -113,7 +110,10 @@ export function useFocusTrap(target: MaybeElementRef, options: UseFocusTrapOptio
     }, { flush: 'post' })
 
   // Cleanup on unmount
-  tryOnUnmounted(deactivate)
+  tryOnUnmounted(() => {
+    stop()
```
Comment (antfu) [inline packages/integrations/useFocusTrap/index.ts:114]: No need for stopping a watch with this hook - it's called automatically.

Comment (wheatjs) [reply packages/integrations/useFocusTrap/index.ts]: Ah right, forgot 😅 

Fixed

File: packages/integrations/useFocusTrap/demo.vue [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+const target = ref()
+const { hasFocus, activate, deactivate } = useFocusTrap(target)
+</script>
+
+<template>
+  <div class="flex flex-col items-center">
+    <button @click="activate()">
+      Trap Focus Within Form
+    </button>
```
Comment (antfu) [inline packages/integrations/useFocusTrap/demo.vue:13]: We could also reflect the activated state on the button

**Pull request merged**